frappe/model/base_document.py

```diff
@@ -345,7 +345,7 @@ def get_parentfield_of_doctype(self, doctype):
 		fieldname = [df.fieldname for df in self.meta.get_table_fields() if df.options==doctype]
 		return fieldname[0] if fieldname else None
 
-	def db_insert(self):
+	def db_insert(self, db_auto_commit = False):
```

Comment:
I don't think this is a good idea. This encourages poor design choices, and I can see how this would lead to data inconsistencies.

Reply:
If we can force the commits to happen upstream appropriately, this can be removed. I just could not figure out a way to do it due to the way the code is setup higher in the stack. To reproduce you can check out the upstream pr and remove the calls where I have set this to true and try running the test suite and see the failures - the ones in india_setup.py were particularly problematic